src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java

```diff
@@ -179,9 +181,27 @@ private Server getServer(RepoFromEnv repoFromEnv) {
         server.setId(repoFromEnv.getId());
         server.setUsername(repoFromEnv.getUsername());
         server.setPassword(repoFromEnv.getPassword());
+        if(repoFromEnv.isUsePreemptiveAuth()) {
+            configurePreemptiveAuth(server);
+        }
         return server;
     }
 
+    private void configurePreemptiveAuth(Server server) {
```

Comment:
This depends on the underlying Maven Resolver version. Modern Maven versions use this approach: https://maven.apache.org/guides/mini/guide-resolver-transport.html

Reply:
i.e. `aether.connector.http.preemptiveAuth` as documented in https://maven.apache.org/resolver-archives/resolver-LATEST-1.x/configuration.html. Can also be setup per repository. Note though that this is not supported for the new JRE based HTTP Client.

Reply:
Looking through https://maven.apache.org/guides/mini/guide-resolver-transport.html#Low-level_Resolver_configuration and https://maven.apache.org/resolver-archives/resolver-LATEST-1.x/configuration.html I adjusted the PR. The property `aether.connector.http.preemptiveAuth.<repoId>` can be used to selectively enable preemptiveAuth for a specific repo